### PR TITLE
Remove deprecated ImageFont.getsize and related functions for Pillow 10.0.0

### PR DESCRIPTION
**Pull request by radarhere**
This removes functions deprecated in #6381.

It includes
- the core change, https://github.com/hugovk/Pillow/pull/99, originally a part of #7059
- adding a new test for `ImageDraw2.textbbox()`, a variation on https://github.com/hugovk/Pillow/pull/100
- only checking width and height of transposed fonts once, a variation on [a suggestion](https://github.com/hugovk/Pillow/pull/100#issuecomment-1498559736) for https://github.com/hugovk/Pillow/pull/100
- an update to preserve backwards compatibility of line spacing by using double the stroke width, one possibility from https://github.com/python-pillow/Pillow/pull/7059#discussion_r1159008846

### CODE REVIEW

**Review by nulano — APPROVED** [review]

File: docs/deprecations.rst [hunk 1]
```diff
@@ -14,55 +14,6 @@ a ``DeprecationWarning`` is issued.
 
 .. _Font size and offset methods:
```
[the remark this answers is not in the document]
Comment (radarhere) [reply docs/deprecations.rst]: The anchor was only being used from ImageDraw documentation that needed to be removed.

I've pushed a commit to remove both.

**Comment by radarhere** [comment]
The Ubuntu failures are also happening on main - https://github.com/python-pillow/Pillow/actions/runs/4656595509

**Comment by hugovk** [comment]
```
Tests/test_qt_image_qapplication.py::test_sanity Fatal Python error: Aborted
```

PyQt6 6.5.0 was released two hours ago: https://pypi.org/project/PyQt6/6.5.0/

**Review by hugovk — APPROVED** [review]

**Comment by radarhere** [comment]
#7081 was merged as a temporary workaround for the Qt failures.

**Pull request merged**